src/gt4py/next/embedded/nd_array_field.py

```diff
@@ -196,9 +196,20 @@ def remap(
 
             # then compute the index array
             xp = self.array_ns
-            new_idx_array = xp.asarray(restricted_connectivity.ndarray) - current_range.start
-            # finally, take the new array
-            new_buffer = xp.take(self._ndarray, new_idx_array, axis=dim_idx)
+            if restricted_connectivity.domain.ndim > 1:
+                offset_abs = [
+                    restricted_connectivity.ndarray
+                    if d == dim
+                    else np.indices(restricted_connectivity.ndarray.shape)[d_i]
+                    for d_i, d in enumerate(new_domain.dims)
+                ]
+                arr_i_abs = np.arange(np.prod(self.shape)).reshape(self.shape)
```

Comment:
to make the 1d index we could use https://numpy.org/doc/stable/reference/generated/numpy.ravel_multi_index.html